client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -0,0 +1,414 @@
+"""Helper functions for load HDA"""
+
+import os
+import contextlib
+import uuid
+
+import ayon_api
+from ayon_api import (
+    get_project,
+    get_representation_by_id,
+    get_versions,
+    get_folder_by_path,
+    get_product_by_name,
+    get_version_by_name,
+    get_representation_by_name
+)
+from ayon_core.pipeline.load import (
+    get_representation_context,
+    get_representation_path_from_context
+)
+from ayon_core.pipeline.context_tools import get_current_project_name
+
+from ayon_core.hosts.houdini.api import lib
+
+import hou
+
+
+def is_valid_uuid(value) -> bool:
+    """Return whether value is a valid UUID"""
+    try:
+        uuid.UUID(value)
+    except ValueError:
+        return False
+    return True
+
+
+@contextlib.contextmanager
+def _unlocked_parm(parm):
+    """Unlock parm during context; will always lock after"""
+    try:
+        parm.lock(False)
+        yield
+    finally:
+        parm.lock(True)
+
+
+def get_available_versions(node):
+    """Return the versions list for node.
+
+    Args:
+        node (hou.Node): Node to query selected products's versions for.
+
+    Returns:
+        List[int]: Version numbers for the product
+    """
+
+    project_name = node.evalParm("project_name") or get_current_project_name()
+    folder_path = node.evalParm("folder_path")
+    product_name = node.evalParm("product_name")
+
+    if not all([
+        project_name, folder_path, product_name
+    ]):
+        return []
+
+    id_only = ["id"]
+    folder_entity = get_folder_by_path(project_name,
+                                       folder_path,
+                                       fields=id_only)
```

Comment:
Please keep formatting consistency. I don't think you need `id_only` variable, especialy when contains only `"id"`.
```suggestion
    folder_entity = get_folder_by_path(
        project_name,
        folder_path,
        fields={"id"})
```